File: java/MASTER_SLAVE_WHOLE_PART_PROXY/DatosCompra.java
import java.util.Date;
import java.util.GregorianCalendar;

public class DatosCompra{
    //Atributos
    private String numeroTarjeta;
    private Date fechaVencimiento;

    //Contructor
    public DatosCompra(){}

    //Métodos
    public void setDatosCompra(String numTarjeta, int dia, int mes, int anio){
        this.numeroTarjeta = numTarjeta;
        GregorianCalendar cal = new GregorianCalendar(anio, mes-1, dia);
        fechaVencimiento = cal.getTime();
    }

    public String getDatosCompra(){
        return "\nINFORMACION DE COMPRA | NUMERO DE TARJETA: " + numeroTarjeta + " | FECHA DE VENCIMIENTO: " + fechaVencimiento; 
    }
}

File: php/MASTER_SLAVE_WHOLE_PART_PROXY/datos/DatosCompra.php
<?php
class DatosCompra{

    private $numeroTarjeta;
    private $fecha;

    public function setDatosCompra($numTarjeta, $fecha){
        $this->numeroTarjeta = $numTarjeta;
        $this->fecha = $fecha;
    }

    public function getDatosCompra(){
        $f = date('d-m-Y', $this->fecha);
        $informacion = $informacion = "<h2>Datos de Compra</h2>";
        $informacion .= "Num de tarjeta: ".$this->numeroTarjeta;
        $informacion .= "<br/>Fecha: ".$f;
        return $informacion;
    }

}